sphinx/ext/todo.py

```diff
@@ -138,10 +138,10 @@ def process_todo_nodes(app, doctree, fromdocname):
 
     for node in doctree.traverse(todolist):
         if not app.config['todo_include_todos']:
-            node.replace_self([])
+            node.replace_self([nodes.target()])
```

Comment:
It seems empty target is generated if not labeled. Then meaningless target will be generated like `<span ...></span>`. of course it is never referenced. I feel it is a bit ugly.

How about add it only if labeled like this?
```
if node.get('ids'):
    node.replace_self([nodes.target()])
else:
    node.replace_self([])
```